src/Util/isWmsLayer.ts

```diff
@@ -0,0 +1,11 @@
+import Layer from 'ol/layer/Layer';
+import Source from 'ol/source/Source';
+import BaseTileLayer from 'ol/layer/BaseTile';
+import BaseImageLayer from 'ol/layer/BaseImage';
+import OlSourceImageWMS from 'ol/source/ImageWMS';
+import OlSourceTileWMS from 'ol/source/TileWMS';
+
+export function isWmsLayer(layer: Layer<Source>): layer is (BaseTileLayer|BaseImageLayer) {
```

Comment:
Is this util ever used? For me this looks something for the `ol-util` package.

Reply:
Yes it needs to be used because the type documentation in openlayers was imprecise in regard of the source types. A PR into openlayers was merged to fix this. But I recognized that I could improve the resulting types of this guard.

Reply:
I still think it would fit better in the `ol-util`. If you want to merge it like this please open an issue to move it. Same goes for the `ArrayTwoOrMore` util.

Reply:
Yes I would say this would be the way to go because `ol-util` is not yet moved to using the ol types

Reply:
I looked at `ol-util` and it is not written in typescript so it does not make sense to add these types there